pkg/kubelet/cm/cgroup_manager.go

```diff
@@ -0,0 +1,104 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cm
+
+import (
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fs"
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+// libcontainerCgroupManager implements CgroupManager interface.
+// It Uses the Libcontainer raw fs cgroup manager for cgroup management.
+type libcontainerCgroupManager struct {
+	// Libcontainer raw fs cgroup manager.
+	fsCgroupManager cgroups.Manager
+}
+
+// Make sure that libcontainerCgroupManager implements the CgroupManager interface
+var _ CgroupManager = &libcontainerCgroupManager{}
+
+// Returns libcontainer's cgroups.config{} struct given the general cgroupConfig
+func getLibcontainerCgroupConfig(cgroupConfig *CgroupConfig) *configs.Cgroup {
+	// Extract the cgroup resource parameters
+	resourceConfig := cgroupConfig.ResourceParameters
+	resources := &configs.Resources{}
+	// skip the parameter if its a zero value
+	if resourceConfig.Memory != 0 {
+		resources.Memory = resourceConfig.Memory
+	}
+	if resourceConfig.CpuShares != 0 {
+		resources.CpuShares = resourceConfig.CpuShares
+	}
+	if resourceConfig.CpuQuota != 0 {
+		resources.CpuQuota = resourceConfig.CpuQuota
+	}
+	// Get the configs.Cgroup object
+	cgroupLibcontainer := &configs.Cgroup{
+		Parent:    cgroupConfig.Parent,
+		Name:      cgroupConfig.Name,
+		Resources: resources,
+	}
+	return cgroupLibcontainer
+}
+
+// Factory Method that returns a configured CgroupManager
+func NewLibcontainerCgroupManager(cgroupConfig *CgroupConfig) *libcontainerCgroupManager {
+	libcontainerCgroupConfig := getLibcontainerCgroupConfig(cgroupConfig)
+	return &libcontainerCgroupManager{
+		fsCgroupManager: &fs.Manager{
+			Cgroups: libcontainerCgroupConfig,
+		},
+	}
+}
+
+// 'Destroy' destroys the associated cgroup set
+func (m *libcontainerCgroupManager) Destroy() error {
+	if err := m.fsCgroupManager.Destroy(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// 'Update' updates the cgroup set with the specified Cgroup Configuration
+func (m *libcontainerCgroupManager) Update(cgroupConfig *CgroupConfig) error {
+	libcontainerCgroupConfig := getLibcontainerCgroupConfig(cgroupConfig)
+	config := &configs.Config{
+		Cgroups: libcontainerCgroupConfig,
+	}
+	if err := m.fsCgroupManager.Set(config); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Create creates the cgroup
+func (m *libcontainerCgroupManager) Create(cgroupConfig *CgroupConfig) error {
+	libcontainerCgroupConfig := getLibcontainerCgroupConfig(cgroupConfig)
+	config := &configs.Config{
+		Cgroups: libcontainerCgroupConfig,
+	}
+	if err := m.fsCgroupManager.Apply(0); err != nil {
```

Comment:
Yes, comments will be appreciated.

BTW, instead of returning the error directly, maybe you'd want to wrap the error to add some more information?